phoenicis-scripts/src/main/java/org/phoenicis/scripts/engine/injectors/IncludeInjector.java

```diff
@@ -21,34 +25,49 @@ public IncludeInjector(ScriptFetcher scriptFetcher) {
 
     @Override
     public void injectInto(PhoenicisScriptEngine phoenicisScriptEngine) {
+        // store scripts that are currently in progress
+        final Stack<String> includeStack = new Stack<>();
         // store included scripts (include path -> JS object)
         final Map<String, Object> includedScripts = new HashMap<>();
 
         phoenicisScriptEngine.put("include", (Function<String, Object>) argument -> {
+            // prevent circular includes
+            if (includeStack.contains(argument)) {
+                throw new CircularIncludeException(argument, includeStack);
```

Comment:
Won't this occur also if a->b->c and a->c?

Reply:
No, because `a -> b -> c` is first evaluated in complete. Only after this returns the second include (i.e. `a -> c`) is evaluated.